app/Http/Transformers/PostTransformer.php

```diff
@@ -48,12 +47,18 @@ public function transform(Post $post)
                 'total' => $post->reactions_count,
             ],
             'status' => $post->status,
-            'location' => $post->location,
             'created_at' => $post->created_at->toIso8601String(),
             'updated_at' => $post->updated_at->toIso8601String(),
         ];
 
-        if (Gate::allows('viewAll', $post)) {
+        // If the post isn't anonymous or if this is the owner of the post, return PII information.
+        if ((! filter_var($post->actionModel->anonymous, FILTER_VALIDATE_BOOLEAN) && ! is_null($post->actionModel->anonymous)) || $post->northstar_id === token()->id()) {
```

Comment:
Instead of using `FILTER_VALIDATE_BOOLEAN`, let's [specify a casting rule](https://laravel.com/docs/5.7/eloquent-mutators#attribute-casting) for this on the Eloquent model.

Reply:
oh i love this!